packages/zero-protocol/src/client-schema.ts

```diff
@@ -0,0 +1,50 @@
+import {mapAllEntries} from '../../shared/src/objects.ts';
+import * as v from '../../shared/src/valita.ts';
+
+export type ValueType = 'string' | 'number' | 'boolean' | 'null' | 'json';
+
+export const valueTypeSchema: v.Type<ValueType> = v.union(
+  v.literal('string'),
+  v.literal('number'),
+  v.literal('boolean'),
+  v.literal('null'),
+  v.literal('json'),
+);
+
+export const columnSchemaSchema = v.object({
+  type: valueTypeSchema,
+});
+
+export type ColumnSchema = v.Infer<typeof columnSchemaSchema>;
+
+export const tableSchemaSchema = v.object({
+  columns: v.record(columnSchemaSchema),
+});
+
+export type TableSchema = v.Infer<typeof tableSchemaSchema>;
+
+export const clientSchemaSchema = v.object({
+  tables: v.record(tableSchemaSchema),
+});
+
+export type ClientSchema = v.Infer<typeof clientSchemaSchema>;
```

Comment:
Idea: should we store the schema used to create permission as part of the permissions data in the DB?  Then we could validate that the permissions schema is compatible with the replicated schema?

If we do, perhaps this should be called ZqlSchema rather than ClientSchema.